Execution_Model_EVM.md

```diff
@@ -0,0 +1,81 @@
+
+Machine state can be thought of as a snap shot of all data held in the EVM at a given moment in time.  Think "state of being."  This includes the aggregate data within the EVM such as account balances and smart contract instances running.  The illustration below depicts the execution state as it changes after a transaction is completed.
+
+![Flow of transactions causing a state change](https://miro.medium.com/max/1282/1*jZ-VRXBJtOnePofB0z2Q8A.png)
+
+# EVM Execution State (Overview)
+
+The execution model of the EVM includes the machine state and the network state. This overview omits the external account (wallet) and associated state as it exists outside of the EVM.  With that, two transaction types exist in the EVM that cause a state change in the execution model:
+
+* Deploy a contract on the Ethereum network
+
+* Execute a function within a deployed contract on the Ethereum Network
+
+# Execution Event: Smart Contract Deployment
+
+When a smart contract has been written and compiled, it is then ready for deployment on the Ethereum blockchain.  This is done by the holder of an externally owned account (i.e. wallet or API) submitting a transaction containing specific information.  The result creates a contract account which will be ready to accept function inputs as well as generate computed and meaningful outputs.
+
+![EOA user deploys new smart contract instance on the Ethereum blockchain](https://i.imgur.com/UmdULIQ.png)
+
+This transaction is constructed as follows:
+
+![Smart contract deployment transaction configuration](https://i.imgur.com/QYepkt5.png)
+
+A developer uses a language such as solidity or vyper to program the smart contract.  Once the high level program code is compiled successfully a bytecode is generated which is a machine code representation of the program.  
+
+![Smart contract code to bytecode](https://i.imgur.com/hPzkXHL.png)
+
+A sent transaction looks like this:
+
+![Send transaction for deploy new smart contract](https://i.imgur.com/Xa7r3h0.png)
+
+We see that a transaction hash is generated (green hash at the bottom.)  Using this information the transaction can be fully examined using eth.getTransaction() and eth.getTransactionReceipt().
+
+![Deployed smart contract transaction receipt](https://i.imgur.com/ZF9raB7.png)
+
+Here we see:
+
+* In the transaction, the to is left empty (‘0x0’ is shown).
+
+* In the input, we only place the bytecode. It is because our SimpleStorage contract does not have a constructor that requires arguments. If arguments are needed in constructor, they are encoded according to the type and appended after the bytecode. We will see how the encoding is done when we call set() function in the next part.
```

Comment:
```suggestion
* In the input, we only place the bytecode. It is because our SimpleStorage contract does not have a constructor that requires arguments. If arguments are needed in constructor, they are encoded according to the type and appended after the bytecode. We will see how the encoding is done when we call `set()` function in the next part.
```